README.md

```diff
@@ -573,6 +573,20 @@ following import:
 ```python
 load("@io_bazel_rules_docker//go:image.bzl", "go_image")
 
+go_image(
+    name = "go_image",
+    srcs = ["main.go"],
+    importpath = "github.com/your/path/here",
+)
+```
+Notice that it is important to explicitly build this target with the
+`--platforms=@io_bazel_rules_go//go/toolchain:linux_amd64` flag
+as the binary should be built for Linux since it will run in a Linux container.
+Only in situations when the `--platforms` flag does not work, try specifying
```

Comment:
do we know now in which situations this does not work? If so lets document them briefly here.

Reply:
I'm actually unaware of any specific cases when the `--platforms` flag will not work.

@jayconrod, from the last part of [your comment](https://github.com/bazelbuild/rules_docker/issues/690#issuecomment-462945896), did you have any specific cases in mind as of when the `--platforms` flag can't be used to cross-compile with rules_go and using the `goos` and `goarch` attrs is inevitable?

From this [doc](https://github.com/bazelbuild/rules_go/blob/master/go/core.rst#attributes-2) about the `goos` and `goarch` attributes it says that:
> This attribute has several limitations and should only be used in situations where the --platforms flag does not work. 

But doesn't mention any cases when the `--platforms` flag may not work.

Reply:
The main situation where `--platforms` doesn't work is where a rule needs to target a specific platform that is different from Bazel's target platform. That mainly comes up in cases where `go_binary` is used in another rule, especially when binaries for multiple platforms are needed.

This use case should be handled by Starlark transitions, but I don't know if they're completely supported yet in Bazel, and rules_go will certainly need an update to take advantage of them.

Reply:
Thank you very much for the explanation.